opendatahub-io/data-science-pipelines-operator
# V1.6.x rhoaieng 1690 and 2385 2

## The issue resolved by this Pull Request:
Resolves for v1.6: 

* https://issues.redhat.com/browse/RHOAIENG-1690
* https://issues.redhat.com/browse/RHOAIENG-2099

## Description of your changes:

This change allows the DSPO to perform heatlh checks against a tls
secured database. If the database is behind a self-signed cert, end user
can provide a ca-bundle either as a global cert (configmap named
"odh-trust-bundle" or via the ".caBundle" config option via the DSPA.

This change also exposes the "ExtraParams" field for DSP, meaning users
can now add ny DSN parameters when configuring a DB connection for DSP.
These params are utilized in the same manner by the DSPO when conducting
the health check to keep the behavior consistent.


## Testing instructions

Deploy a secure/tls enabled s3/mariadb behind a self-signed cert (for example in a self-signed ocp cluster), provide these configs to the DSAP as external configs.  Ensure they only accept tls based connections.

Deploy DSPA configured to leverage these connections via external db/object store connections. 

Confirm DSPO is able conduct successfull health checks (if it deploys the DSPA pods then health checks worked), if it didn't, log the error here. 

Confirm DSPA comes up successfully, in this change DSPO willl udpate the DSP server's configs to enable "tls" for db connections. Confirm this behavior by running a successful pipeline. 



## Checklist
